### ISSUE TITLE
Binary Firebase Frameworks for tvOS

### ISSUE THREAD

**Issue by dpott197** [firebase/firebase-ios-sdk#4550]
Describe your environment

  * Xcode version: 11.3
  * Firebase SDK version: 6.14
  * Firebase Component: Analytics
  * Component version: 6.14

Describe the problem

#### Steps to reproduce:

Are there Firebase Frameworks specifically built for tvOS like https://firebase.google.com/download/ios? Right now, I am trying to use Firebase without CocoaPods. I have been successfully linking Crashlytics and Fabric frameworks without issue. When I attempt to link Firebase, I get the following error...

```
ld: in FirebaseAnalytics/FirebaseCore.framework/FirebaseCore(FIRAnalyticsConfiguration.o), building for tvOS, but linking in object file (FirebaseAnalytics/FirebaseCore.framework/FirebaseCore(FIRAnalyticsConfiguration.o)) built for iOS, file 'Firebase/ios/FirebaseAnalytics/FirebaseCore.framework/FirebaseCore' for architecture arm64
```

#### Relevant Code:

```
        #-----------------------------------------------------------------
        # Include FirebaseCore Framework
        set(FIREBASE_CORE_FRAMEWORK_PATH ${PATH_TO_FIREBASE_ANALYTICS}/FirebaseCore.framework)
        set(FIREBASE_CORE_HEADERS_PATH ${FIREBASE_CORE_FRAMEWORK_PATH}/Headers)

        add_library(firebasecore STATIC IMPORTED)
        set_property(TARGET firebasecore PROPERTY IMPORTED_LOCATION ${FIREBASE_CORE_FRAMEWORK_PATH}/FirebaseCore)
        target_link_libraries(${PROJECT_NAME} PUBLIC firebasecore)
        include_directories(SYSTEM ${FIREBASE_CORE_HEADERS_PATH})
```

**Comment by paulb777** [comment]
We don't support binary distributions for tvOS.  

Details about tvOS support at https://github.com/firebase/firebase-ios-sdk#tvos-macos-and-catalyst

**Comment by jasonhowk-mlb** [comment]
So this should also rule out using the google provided Carthage binaries then on tvOS as well, correct?

**Comment by paulb777** [comment]
Yep, same story for Carthage. However, it might not be too hard to add tvOS support to the ZipBuilder [here](https://github.com/firebase/firebase-ios-sdk/tree/master/ZipBuilder) and we would accept PRs.

**Comment by vijayviswas** [comment]
+1

**Comment by gangadharbonda** [comment]
We also require the similar support for tvOS and trying to look for options but could not succeed. At last, I tried to add the project files and then tried to move on

**Comment by vijayviswas** [comment]
@gangadharbonda  Can explain in more detail like which are the framework source files needed to be added to tvos project?

**Comment by vijayviswas** [comment]
@paulb777  Is there any progress on this as some of my client tVOS apps cant leverage crash reporting due to this

**Comment by paulb777** [comment]
@vijayviswas We don't have near term plans for this.  The best near term approach would be to build it yourself. We would consider a PR as well.

**Comment by vijayviswas** [comment]
@paulb777  - Is the pod for analytics for tvOS? I tried below in tvOS cocopods and it doesnt install any pods. All other pods are installed. As there are no "Analytics" pods I dont see any "crash free user" data in tvOS firebase crashlytics portal.
pod 'Firebase/Analytics'

**To see this data, make sure your app has Google Analytics for Firebase and the latest Firebase Crashlytics SDK.**

**Comment by JavanPoirier** [comment]
Hello folks, I got Crashlytics working on tvOS without Cocoapods or XCFramework bundles. You can read the story here: https://medium.com/@Javan.Poirier/integrating-firebase-crashlytics-for-tvos-without-cocoapods-or-xcframework-support-a8f4b389c7c2

**Comment by paulb777** [comment]
We're planning to add tvOS and macOS to the zip distribution in the next release for the Firebase products that support those platforms.

**Comment by dpott197** [comment]
Thank you!

**Comment by paulb777** [comment]
Available in today's 7.4.0 release - https://github.com/firebase/firebase-ios-sdk/releases/tag/7.4.0
